### PR TITLE
[Feature Store] Check if mapping is unique and copy data before saving to nosql target

### PR DESCRIPTION
**Pull request by davesh0812**
This pr address to [ML-2719](https://jira.iguazeng.com/browse/ML-2719).

Also copy data before saving to nosql target, because when saving df to nosql target it's changed unexpectedly.

### CODE REVIEW

File: mlrun/datastore/targets.py [hunk 1]
```diff
@@ -1115,6 +1115,7 @@ def write_dataframe(
             df = self.prepare_spark_df(df)
             df.write.mode("overwrite").save(**options)
         else:
+            df = df.copy(deep=False)
```
Comment (assaf758) [inline mlrun/datastore/targets.py:1118]: please add a comment why you are doing a copy.
also, is shallow copy enough?

**Review by assaf758 — CHANGES_REQUESTED** [review]
looks good, but can you extend OneHotEncoder test to cover this?

File: mlrun/datastore/targets.py (hunk 1 above)
Comment (davesh0812) [reply mlrun/datastore/targets.py:1119]: @assaf758  I think that In most cases shallow copy is enough

**Review by assaf758 — APPROVED** [review]

File: mlrun/datastore/targets.py [hunk 2]
```diff
@@ -1115,6 +1115,8 @@ def write_dataframe(
             df = self.prepare_spark_df(df)
             df.write.mode("overwrite").save(**options)
         else:
+            # To prevent modification of the original dataframe
+            df = df.copy(deep=False)
```
Comment (gtopper) [inline mlrun/datastore/targets.py:1119]: Is this actually a problem? It doesn't look like `frames_client.write` modifies the df passed to it.

Comment (davesh0812) [reply mlrun/datastore/targets.py:1119]: @gtopper you can run and see it is modifies the df

Comment (gtopper) [reply mlrun/datastore/targets.py:1119]: Opened https://jira.iguazeng.com/browse/IG-21310 for this issue.

**Review by gtopper — APPROVED** [review]
Opened https://jira.iguazeng.com/browse/IG-21310 for the v3io-frames issue.

**Pull request merged**